### PR TITLE
[`GemmaConverter`] use user_defined_symbols

### LINKED ISSUE #29440: AutoTokenizer tokenization issue
### System Info

On colab, transformers==4.38.2

### Who can help?

@ArthurZucker @younesbelkada

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
from transformers import GemmaTokenizer, AutoTokenizer

gemma_tokenizer = GemmaTokenizer.from_pretrained("google/gemma-7b")
gemma_auto_tokenizer = AutoTokenizer.from_pretrained("google/gemma-7b")

print(gemma_tokenizer.tokenize("<start_of_turn>"))
print(gemma_auto_tokenizer.tokenize("<start_of_turn>"))
```
The result is
```
['<start_of_turn>']
['<', 'start', '_', 'of', '_', 'turn', '>']
```

### Expected behavior

Both tokenization result should be same.

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes #29440 by always adding the `user_defined_symbols` to the tokenizer, otherwise because no merges exist for them, they will be split.

### CODE REVIEW

**Review by pcuenca — APPROVED** [review]
Great idea! There are a few additional tokens in`user_defined_symbols`, I'll push PRs to the repos.

**Comment by ArthurZucker** [comment]
I will add a test and we can merge

File: tests/models/llama/test_tokenization_llama.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
 @require_sentencepiece
 @require_tokenizers
 class LlamaTokenizationTest(TokenizerTesterMixin, unittest.TestCase):
-    from_pretrained_id = "hf-internal-testing/llama-tokenizer"
+    from_pretrained_id = ["hf-internal-testing/llama-tokenizer", "meta-llama/Llama-2-7b-hf"]
```
Comment (LysandreJik) [inline tests/models/llama/test_tokenization_llama.py:55]: Looks good to me!

**Review by LysandreJik — APPROVED** [review]
Understood! cc @ydshieh that this will introduce some failing tests in the slow CI, we're aware (maybe we can skip those until we get to it? after the release)

**Comment by ydshieh** [comment]
Thank you for warning me. But could you share with me the reason to merge before we resolve the failing tests?

**Comment by ArthurZucker** [comment]
We need to release !

**Comment by ArthurZucker** [comment]
We can open a PR to skip / fix them tomorrow!

**Pull request merged**